reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/WithLatestFrom.kt

```diff
@@ -26,7 +24,8 @@ fun <T, U, R> Observable<T>.withLatestFrom(
         emitter.setDisposable(disposables)
 
         val otherSources = others.toList()
-        val otherValues = atomicList<Any?>(List(otherSources.size) { Uninitialized })
+        val otherValues = HashMap<Int, U>()
```

Comment:
Maybe 2 arrays instead (lastValue + isInitialized)? Should be both faster and more memory efficient.

Reply:
This would require iterating over the `isInitialized` array every time in the upstream's `onNext` method? Or we would need to also memoize it with a flag. WDYT?

Reply:
Fixed, please review